File: api/src/models/OptionModel.py
from typing import List, TYPE_CHECKING
import uuid
from database import Base
from sqlalchemy import Boolean, ForeignKey, Integer,  String, UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship



if TYPE_CHECKING:
    from .QuestionModel import Question
    from .AnswerModel import Answer


class Option(Base):
    __tablename__ = "options"


    id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True, index=True, default=uuid.uuid4)
    description: Mapped[str] = mapped_column(String(250), nullable=False)
    points: Mapped[int] = mapped_column(Integer, nullable=True)



    question_id: Mapped[uuid.UUID] = mapped_column(UUID, ForeignKey("questions.id"), nullable=False)



    question: Mapped["Question"] = relationship(back_populates="options", lazy="selectin")
    answers: Mapped[List["Answer"]] = relationship(back_populates="option", cascade="all, delete", lazy="selectin")

File: client/surbased/lib/src/survey/domain/option_model.dart
class Option {
  final String? description;
  final int? points;
  final String? id;
  final String? questionId;

  Option({
    this.description,
    this.points,
    this.id,
    this.questionId,
  });

  factory Option.fromJson(Map<String, dynamic> json) => Option(
        description: json["description"],
        points: json["points"],
        id: json["id"],
        questionId: json["question_id"],
      );

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> json = {};
    if (description != null) json["description"] = description;
    if (points != null) json["points"] = points;
    if (id != null) json["id"] = id;
    if (questionId != null) json["question_id"] = questionId;
    return json;
  }
}
